aroquev/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,35 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> comments;
+
+  @Override
+  public void init() {
+    comments = new ArrayList<>();
+    comments.add("I would like to live in NYC.");
+    comments.add("Berlin looks like a nice place to have fun.");
+    comments.add("I gotta go visit Japan one day!");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Armando!</h1>");
+    // Convert the comments to JSON
+    String json = convertToJson(comments);
+
+    // Send the JSON as response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
   }
+
+  /**
+   * Convert to JSON using Gson
+   */
+  private String convertToJson(ArrayList comments) {
```

Comment:
Add type parameter. References to generic types should be parameterized.

```java
private String convertToJson(ArrayList<String> comments) {...}
```

Reply:
Done.